File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] mat) {
        int n = mat.length;
        int m = mat[0].length;
        boolean row = false, col = false;

        for(int i = 0; i<n; i++) {
            for(int j = 0; j<m; j++) {
                if(mat[i][j] == 0) {
                    if(i == 0) row = true;
                    if(j == 0) col = true;
                    mat[i][0] = 0;
                    mat[0][j] = 0;
                }
            }
        }

        for(int i = 1; i<n; i++) {
            for(int j = 1; j<m; j++) {
                if(mat[i][0] == 0 || mat[0][j] == 0) mat[i][j] = 0;
            }
        }

        if(row) {
            for(int j = 0; j<m; j++) {
                mat[0][j] = 0;
            }
        }

        if(col) {
            for(int i = 0; i<n; i++) {
                mat[i][0] = 0;
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        n, m = len(matrix), len(matrix[0])
        row, col = 0, 0
        
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    if i == 0:
                        row = 1
                    if j == 0:
                        col = 1
                    matrix[i][0] = 0
                    matrix[0][j] = 0
                
                
        for i in range(1, n):
            for j in range(1, m):
                if matrix[i][0] == 0 or matrix[0][j] == 0:
                    matrix[i][j] = 0
        
        if row == 1:
            for j in range(m):
                matrix[0][j] = 0
                
        if col == 1:
            for i in range(n):
                matrix[i][0] = 0
        